repository/src/main/java/com/epam/esm/repository/impl/AbstractRepository.java

```diff
@@ -0,0 +1,44 @@
+package com.epam.esm.repository.impl;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Repository
+public abstract class AbstractRepository<T> {
```

Comment:
Please use beans annotations in appropriate places.

Reply:
agree with you, fixed